### PR TITLE
Create edge function to get the permission of user in a room

### LINKED ISSUE #83: Create a edge function to return the permission of user in a room
Create a edge function to return the permission of user in a room

### PR DESCRIPTION
**Pull request by hernanr99**
Create edge function called get-room-permission to get the permission of user in a room

### CODE REVIEW

File: backend/src/supabase/functions/get-room-permission/index.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
+import { serve } from 'https://deno.land/std@0.131.0/http/server.ts';
+import { createClient } from 'https://esm.sh/@supabase/supabase-js@2';
+import { corsHeaders } from '../_shared/cors.ts';
+
+async function getPermission(supabaseClient, body) {
+  let { userId, roomId }= body;
```
Comment (marcovidonis) [inline backend/src/supabase/functions/get-room-permission/index.ts]:
```suggestion
  let { userId, roomId } = body;
```

File: backend/src/supabase/functions/get-room-permission/index.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    return new Response(JSON.stringify({
+      error: 'No room or user with given data exists'
+    }), {
+      headers: {
+        ...corsHeaders,
+        'Content-Type': 'application/json'
+      },
+      status: 404
+    });
+  }
```
Comment (marcovidonis) [inline backend/src/supabase/functions/get-room-permission/index.ts]: Question: which is the best way to handle this case? If we have no result here, it just means that this user has no special permissions in this room: it's not technically an error.

If we send a 404 error here, we'll likely need to handle it separately on the Front-End. Alternatively, we could send a 200 response with `permission: null`. Then, on the Front-End, we'll set up a logic like `if (!res.data.permission) { // user is guest }`.

What do you think?

Comment (nreyes-qualabs) [reply backend/src/supabase/functions/get-room-permission/index.ts]: I agree with @marcovidonis here. 404 should be returned if for example the user or the room don't exist to begin with. But the user having no roles is an expected possible result that should return 200. That may be what the code is already doing though, I'm not sure by just looking at it

File: backend/src/supabase/functions/get-room-permission/index.ts [hunk 3]
```diff
@@ -2,27 +2,42 @@ import { serve } from 'https://deno.land/std@0.131.0/http/server.ts';
...
 
+async function returnError(msgError: string){
+  return new Response(JSON.stringify({
+    error: msgError
+  }), {
+    headers: {
+      ...corsHeaders,
+      'Content-Type': 'application/json'
+    },
+    status: 400
```
Comment (marcovidonis) [inline backend/src/supabase/functions/get-room-permission/index.ts:13]: You use `returnError` when you can't find the room or the user, so you should probably return a 404 here.
Alternatively, to make it more generic, you could pass a second parameter to this function where you specify the status code.

File: backend/src/supabase/functions/get-room-permission/index.ts [hunk 4]
```diff
@@ -2,27 +2,42 @@ import { serve } from 'https://deno.land/std@0.131.0/http/server.ts';
...
+
+  const users = await supabaseClient.from('users-data').select('userId').eq('userId', userId);
+  console.error(users);
+  if (!users.data || (users.data && users.data.length === 0))
+    return returnError("No user with given data exists");
+
+  const roomsData = await supabaseClient.from('rooms-data').select('permissionId').match({ roomId, userId });
+
+  let permission = "GUEST"
+  if (!roomsData.data || (roomsData.data && roomsData.data.length !== 0)){
```
Comment (marcovidonis) [inline backend/src/supabase/functions/get-room-permission/index.ts]: I think there's something wrong here... perhaps do something like:
```suggestion
  if (roomsData.data?.length) {
```

**Review by marcovidonis — CHANGES_REQUESTED** [review]

**Review by marcovidonis — APPROVED** [review]

**Pull request merged**